pkg/server/server.go

```diff
@@ -165,6 +168,35 @@ func getClientConfigRequest(r *http.Request) (*config.ConfigRequest, error) {
 	return configRequest, nil
 }
 
+// retrieve the UuidRequest request
+func getClientUuidRequest(r *http.Request) (*uuid2.UuidRequest, error) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Body read failed: %v", err)
+		return nil, err
+	}
+	uuidRequest := &uuid2.UuidRequest{}
+	err = proto.Unmarshal(body, uuidRequest)
+	if err != nil {
+		log.Printf("Unmarshalling failed: %v", err)
+		return nil, err
+	}
+	return uuidRequest, nil
+}
+
+//parseX509Cert parses incoming certificate
+func parseX509Cert(inp []byte) (*x509.Certificate, error) {
```

Comment:
Either the cert needs to be removed from the uuid message and you get the cert from the authcontainer, or you need to check that the cert in the uuid message and the authcontainer are identical.

Reply:
Coming back to the earlier conversation, isn't authcontainer an entirely v2 construct? We should use defer on this until we get adam to v2.